query/sources/impl/src/main/java/org/codice/ddf/admin/sources/opensearch/OpenSearchSourceUtils.java

```diff
@@ -59,96 +60,102 @@
+        UrlField requestUrl = responseField.requestUrlField();
+
+        if (statusCode != HTTP_OK || responseBody.length() < 1) {
+            addUnknownEndpointError(configResult, urlField);
+            return configResult;
         }
 
         Document capabilitiesXml;
         try {
-            capabilitiesXml = createDocument(responseBodyResult.result());
+            capabilitiesXml = sourceUtilCommons.createDocument(responseBody);
         } catch (Exception e) {
             LOGGER.debug("Failed to read response from OpenSearch endpoint.");
-            responseBodyResult.addArgumentMessage(unknownEndpointError(urlField.path()));
-            return responseBodyResult;
+            addUnknownEndpointError(configResult, urlField);
+            return configResult;
         }
 
         XPath xpath = XPathFactory.newInstance()
                 .newXPath();
         xpath.setNamespaceContext(SOURCES_NAMESPACE_CONTEXT);
+
         try {
             if ((Boolean) xpath.compile(TOTAL_RESULTS_XPATH)
                     .evaluate(capabilitiesXml, XPathConstants.BOOLEAN)) {
-                return responseBodyResult;
+                OpenSearchSourceConfigurationField config =
+                        new OpenSearchSourceConfigurationField();
+                config.endpointUrl(responseField.requestUrl())
+                        .credentials()
+                        .username(creds.username())
+                        .password(FLAG_PASSWORD);
+
+                configResult.result(config);
             }
         } catch (XPathExpressionException e) {
             LOGGER.debug("Failed to compile OpenSearch totalResults XPath.");
-            responseBodyResult.addArgumentMessage(unknownEndpointError(urlField.path()));
-            return responseBodyResult;
-
+            addUnknownEndpointError(configResult, urlField);
         }
-        responseBodyResult.addArgumentMessage(unknownEndpointError(urlField.path()));
-        return responseBodyResult;
+
+        return configResult;
     }
 
-    /**
-     * Attempts to discover an OpenSearch endpoint from the given hostname and port
-     *
-     * Possible Error Codes to be returned
-     * - {@link org.codice.ddf.admin.common.report.message.DefaultMessages#CANNOT_CONNECT}
-     *
-     * @param hostField hostname and port to probe for OpenSearch capabilities
-     * @param creds        optional credentials for authentication
-     * @return a {@link ReportWithResultImpl} containing the discovered {@link UrlField} on success, or containing {@link org.codice.ddf.admin.api.report.ErrorMessage}s on failure.
-     */
-    public ReportWithResultImpl<UrlField> discoverOpenSearchUrl(HostField hostField, CredentialsField creds) {
-        return requestUtils.discoverUrlFromHost(hostField, URL_FORMATS, creds,
-                GET_CAPABILITIES_PARAMS);
+    private void addUnknownEndpointError(ReportWithResultImpl report, UrlField urlField) {
```

Comment:
When the discovery is being done with host name and port, the url will be null will be null. In that case I don't want a path. I agree this sucks though. I didn't want to spend too much time on this, but will look again.